credentials/apps/badges/models.py

```diff
@@ -149,6 +149,12 @@ def save(self, *args, **kwargs):
             super().save(*args, **kwargs)
         else:
             raise ValidationError("Cannot update BadgeRequirement for active BadgeTemplate")
+    
+    def reset(self, username: str):
+        try:
+            Fulfillment.objects.get(progress__username=username, progress__template=self.template).delete()
```

Comment:
It's possible to catch `multiple objects returned` - there are cases where the Badge template progress has more than 1 Fulfillment (multiple requirements fulfilled).

And we don't want to reset all the progress but only a single requirement completion. 

```
# maybe:
Fulfillment.objects.filter(requirment=self, progress__username=username, progress__template=self.template).delete()
```
Return <bool> based on deletion result (e.g. if reset has happened).

Reply:
Let's also add the whole progress reset:

```
badge_progress.reset(username) >> removes all related Fulfillment items
```